session/session.go

```diff
@@ -2928,8 +2928,14 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
 	if dom.GetEtcdClient() != nil {
 		// We only want telemetry data in production-like clusters. When TiDB is deployed over other engines,
 		// for example, unistore engine (used for local tests), we just skip it. Its etcd client is nil.
-		dom.TelemetryReportLoop(ses[5])
-		dom.TelemetryRotateSubWindowLoop(ses[5])
+		if config.GetGlobalConfig().EnableTelemetry {
+			// There is no way to turn telemetry on with global variable `tidb_enable_telemetry`
+			// when it is disabled in config. See IsTelemetryEnabled function in telemetry/telemetry.go
+			go func() {
```

Comment:
Is adding `go func()` here like this not changing the behaviour? Is this ok for a bugfix release?

Reply:
Hm, it shouldn't affect feature but just make initialization synchronous and speed up startup when connecting to telemetry server is blocked by network policy.

Reply:
The reason to add `go` here was I realized that starting tidb-server takes about 20 seconds when the hosting environment doesn't have internet access.

Reply:
But I think you are right. We should keep changes as small as possible for back port. I have reverted this according to the behavior in 6.1. PTAL